.buildbot.sh

```diff
@@ -8,3 +8,32 @@ set -e
 ulimit -d $((1024 * 1024 * 8)) # 8 GiB
 
 ./x.py test --stage 1 --exclude rustdoc-json --exclude debuginfo
+
+COMMIT_HASH=$(git rev-parse --short HEAD)
+TARBALL_TOPDIR=`pwd`/build/rustgc-stage2-latest
+TARBALL_NAME=rustgc-${CI_TRACER_KIND}-stage2-${COMMIT_HASH}.tar.bz2
+SYMLINK_NAME=rustgc-${CI_TRACER_KIND}-stage2-latest.tar.bz2
+SNAP_DIR=/opt/rustgc-bin-snapshots
+
+/usr/bin/time -v ./x.py test --config .buildbot.config.toml --stage 2 --exclude rustdoc-json --exclude debuginfo
+
+# Build extended tools and install into TARBALL_TOPDIR.
+mkdir -p ${TARBALL_TOPDIR}
+/usr/bin/time -v ./x.py install --config .buildbot.config.toml
+
+# Check that the install looks feasible.
+for i in rustc cargo rustdoc; do
+    test -e ${TARBALL_TOPDIR}/bin/${i}
+done
+
+# Archive the build and put it in /opt
+git show -s HEAD > ${TARBALL_TOPDIR}/VERSION
+cd build
+tar jcf ${TARBALL_NAME} `basename ${TARBALL_TOPDIR}`
+chmod 775 ${TARBALL_NAME}
+mv ${TARBALL_NAME} ${SNAP_DIR}
+ln -sf ${SNAP_DIR}/${TARBALL_NAME} ${SNAP_DIR}/${SYMLINK_NAME}
+
+# Remove all but the 10 latest builds
+cd ${SNAP_DIR}
+sh -c "ls -tp | grep -v '/$' | tail -n +11 | xargs -I {} rm -- {}"
```

Comment:
I suggest we keep fewer builds (with ykrustc we've never made use of this feature): say 3? So it would be `tail -n +4`.

Reply:
Sure. Fixed 55f3fa3